github_test.go

```diff
@@ -0,0 +1,370 @@
+			mustWriteJSON(w, github.RepositoryContentResponse{
+				Content: &github.RepositoryContent{
+					Type:     github.String("file"),
+					Encoding: github.String("base64"),
+					Size:     github.Int(len(base64Encoded(resultsBytes))),
+					Name:     github.String(path),
+					Path:     github.String(filepath.Join(specimen.Dir, path)),
+					Content:  github.String(base64Encoded(resultsBytes)),
+					SHA:      github.String(sha(resultsBytes)),
+				},
+				Commit: github.Commit{
+					SHA: github.String(serverSHAForRepo),
+				},
+			})
+			return
+		}
+		if r.Method == "DELETE" && checkFileRegexp.MatchString(path) {
+			// 1. Enforce body contains message, committer, content (base64-encoded), sha
+			stuff := struct {
+				Message   string            `json:"message"`
+				Content   string            `json:"content"`
+				SHA       string            `json:"sha"`
+				Committer map[string]string `json:"committer"`
+			}{}
+			if err := json.NewDecoder(r.Body).Decode(&stuff); err != nil {
+				t.Errorf("Expected body to decode fine, but got %+v", err)
+			}
+			if expected := fmt.Sprintf("[checkup] delete %s [ci skip]", path); stuff.Message != expected {
+				t.Errorf("Expected commit message to be '%s', got '%s'", expected, stuff.Message)
+			}
+			if stuff.SHA != serverSHAForRepo {
+				t.Errorf("Expected SHA to be %s, got '%s'", serverSHAForRepo, stuff.SHA)
+			}
+			if stuff.Committer["email"] != specimen.CommitterEmail {
+				t.Errorf("Expected email to be %s, got %s", specimen.CommitterEmail, stuff.Committer["email"])
+			}
+			if stuff.Committer["name"] != specimen.CommitterName {
+				t.Errorf("Expected name to be %s, got %s", specimen.CommitterName, stuff.Committer["name"])
+			}
+
+			// Ok, start modifying the in-memory state.
+			if _, ok := index[path]; !ok {
+				http.Error(w, "no such file: "+path, 500)
+			}
+			delete(index, path)
+
+			if _, ok := gitRepo.Files[filepath.Join(specimen.Dir, path)]; !ok {
+				http.Error(w, "file was deleted", 401)
+			}
+			gitRepo.Files[filepath.Join(specimen.Dir, path)] = false
+			gitRepo.LastUpdated = time.Now().UnixNano()
+			serverSHAForRepo = sha(toJSON(gitRepo))
+
+			// Response is the same if updating or creating.
+			mustWriteJSON(w, github.RepositoryContentResponse{
+				Commit: github.Commit{
+					SHA: github.String(serverSHAForRepo),
+				},
+			})
+			return
+		}
+		http.Error(w, path+" is not handled", 403)
+		t.Errorf("Cannot handle %s %s (path=%s)", r.Method, r.URL.Path, path)
+	})
+
+	// Here goes!
+
+	if err := specimen.Store(results); err != nil {
+		t.Fatalf("Expected no error from Store(), got: %v", err)
+	}
+
+	fmt.Println("Done with Store()")
+
+	// Make sure index has been created
+	index, _, err := specimen.readIndex()
+	if err != nil {
+		t.Fatalf("Cannot read index: %v", err)
+	}
+
+	if len(index) != 3 {
+		t.Fatalf("Expected length of index to be 3, but got %v", len(index))
+	}
+
+	var (
+		name string
+		nsec int64
+	)
+	for name, nsec = range index {
+	}
```

Comment:
You could also add a `break` inside the `for` loop. No point in iterating over all entries until the last, when the first is just as good.